detector/muonSystem/muonSystemMuRWELL_o1_v01.cpp

```diff
@@ -0,0 +1,941 @@
+      dd4hep::Box sideEnvelope2(sideEnvX, sideEnvY2, sideEnvZ);      
+      std::string sideName = dd4hep::xml::_toString(side, "side%d");
+      dd4hep::Volume sideVol(sideName, sideEnvelope, mat);
+      dd4hep::Volume sideVol2(sideName, sideEnvelope2, mat);      
+
+      double angle_degrees = shapeAngle * side; // Calculate the angle for each side
+      double angle_radians = angle_degrees * M_PI / 180.0;
+
+      double sideXOffset = radius * std::cos(angle_radians+shapeAngle_radians);
+      double sideYOffset = radius * std::sin(angle_radians+shapeAngle_radians);
+
+      double sideXOffset2 = (radius + detectorVolumeThickness) * std::cos(angle_radians+shapeAngle_radians);
+      double sideYOffset2 = (radius + detectorVolumeThickness) * std::sin(angle_radians+shapeAngle_radians);      
+
+      dd4hep::RotationZ sideRotationZ(angle_radians+shapeAngle_radians+angle_clearance);
+  //    dd4hep::RotationY sideRotationY(90.* dd4hep::degree);     
+      dd4hep::Rotation3D sideRotation = dd4hep::Rotation3D(sideRotationZ);      
+
+      double sideXPos = sideXOffset ; 
+      double sideYPos = sideYOffset ;   
+      double sideZPos = 0.0; 
+      double sideXPos2 = sideXOffset2 ; 
+      double sideYPos2 = sideYOffset2 ;       
+
+      dd4hep::Position sideTrans;
+      dd4hep::PlacedVolume sidePhys;
+      dd4hep::DetElement sideDE;
+
+      if (side % 2 == 0) {
+        sideTrans = dd4hep::Position(sideXPos, sideYPos, sideZPos); // Assign value inside if block
+        sidePhys = BarrelDetectorLayerVolume.placeVolume(sideVol, dd4hep::Transform3D(sideRotation, sideTrans));
+        sideDE = dd4hep::DetElement(detElement, sideName + "DE", side);
+        sideDE.setPlacement(sidePhys);
+        sideVol.setVisAttributes(lcdd, xmlDet.visStr());  
+
+      } else {
+        sideTrans = dd4hep::Position(sideXPos2, sideYPos2, sideZPos); // Assign value inside else block
+        sidePhys = BarrelDetectorLayerVolume.placeVolume(sideVol2, dd4hep::Transform3D(sideRotation, sideTrans));
+        sideDE = dd4hep::DetElement(detElement, sideName + "DE", side);
+        sideDE.setPlacement(sidePhys);
+        sideVol2.setVisAttributes(lcdd, xmlDet.visStr());        
+      } 
+
+// -------------------------------------------------------------------------------------------------------
+//  Dividing every side to small rectangles  //
+// -------------------------------------------------------------------------------------------------------
+
+      int numRectangles = barrelLength / (2 * dimensions.z() - overlapZ); // numbers of the rectangles in each side.. depends on the number of the chambers that can be overlapped in Z direction
+        
+      for (int rectangle = 0; rectangle < (numRectangles + 1); ++rectangle) {
+
+          double rectangleEnvX = detectorVolumeThickness/4.5;  // dividing by 4.5 gets the best thickness for the recatngle to avoid any overlap ~ in our case the uRWELL the best rectangle thickness is 26.667 mm which is 120/4.5.
```

Comment:
This  `/4.5` seems awfully fine-tuned? Or is this OK also for other thinknesses?

Reply:
Hi @andresailer, I have changed the parameters, and now it is OK with other thicknesses to. Whatever the chamber dimensions the user will put in the compact file, It will work. I have made the variable  `detectorVolumeThickness` is determined by the builder depending on the chamber dimensions and the overlap angle, instead before it was determined by the user.